roundedimageview/src/main/java/com/makeramen/roundedimageview/RoundedImageView.java

```diff
@@ -266,7 +278,45 @@ private Drawable resolveResource() {
 
   @Override
   public void setBackground(Drawable background) {
-    setBackgroundDrawable(background);
+      mBackgroundDrawable = background;
+      updateBackgroundDrawableAttrs(true);
+      if(BuildConfig.VERSION_CODE >= Build.VERSION_CODES.JELLY_BEAN)
```

Comment:
Even the source of View.java just proxies the deprecated function: https://github.com/android/platform_frameworks_base/blob/master/core/java/android/view/View.java#L17225

Let's keep that behavior and put our logic inside `setBackgroundDrawable(...)` instead of trying to reverse it here.